### ISSUE TITLE
Sandbox support for http2

### ISSUE THREAD

**Issue by kristoferjoseph** [arc-repos/architect#236]
## Reason
This is a preliminary feature tracking issue for using http2 locally with Architect sandbox.

The main reason for this feature is to be able to better replicate real world scenarios and allow users to try out http2 features in their local environment.

One http2 feature we rely on in production is the ability to load resources in parallel which is enabled by http2.

Another feature I personally am excited to start using is http push.

## Caveats

Some caveats that will need to be considered with this feature are as follows:
1. Architect sandbox will need to be updated to use the experimental node http2 module to create an http1 compatible secure server.
*see example change set below

2. Users will need to generate a key pair for their machine's localhost path. This can be accomplished with a script that uses openssl or similar to generate a cert and key file that is specific to the user's machine and is not checked in.
```
openssl req -x509 -newkey rsa:2048 -nodes -sha256 -subj '/CN=localhost' \
  -keyout localhost-privkey.pem -out localhost-cert.pem
```

3. A user will need to access localhost via `https` 
  a. To enable this the user would need to generate a self signed or legitimate cert for localhost or accept a permanent exception in their browser.
`https://localhost:3333`

The minimal perspective change set in `/src/sandbox/http/index.js` could look something like this:
```
var http = require('http2')
...
// securitay
const cert = fs.readFileSync(path.join(__dirname, './localhost-cert.pem'))
const key = fs.readFileSync(path.join(__dirname, './localhost-privkey.pem'))
const serverOptions = {
  cert,
  key,
  allowHTTP1: true
}
...
  server = http.createSecureServer(serverOptions, function _request(req, res) {
    app(req, res, finalhandler(req, res))
  })
...
```

Curious to hear thoughts on this.

**Comment by ryanblock** [comment]
This is dope. Maybe this should be a sandbox cli flag (`--http2`) so we know whether to look for this? Also makes it opt-in in case of any bugs.

**Comment by filmaj** [comment]
Not sure whether to comment in here or the PR so I'll keep it to here.

I definitely approve of the idea, makes complete sense. However, I think the implementation / delivery via arc needs some thought. The current approach of generating a key pair using an `npm run` script won't work for consumers of just arc. E.g. for apps I'm building, I had to run the script to generate keys from the arc repo and copy them into each of my apps. If I didn't do that, I'd see:

```
➜ npx sandbox
ENOENT: no such file or directory, open './localhost-cert.pem'
(node:6308) ExperimentalWarning: The http2 module is an experimental API.
```

I also saw the following warning when playing around with some of my apps:

```
(node:6859) UnsupportedWarning: Status message is not supported by HTTP/2 (RFC7540 8.1.2.4)
```

Not sure where that is coming from...

Some other thoughts:

1. The localhost dev experience does suck a bit more w/ having to accept the warning, especially the first-time developer experience. If I understand correctly (via help from [this letsencrypt article](https://letsencrypt.org/docs/certificates-for-localhost/) and [this medium article](https://medium.freecodecamp.org/how-to-get-https-working-on-your-local-development-environment-in-5-minutes-7af615770eec)), to avoid this, we'd need to:
    1. Generate a root cert
    2. Issue a cert using this root
    3. Make sure the local dev environment imports the generated root cert into its local authorized cert store.

To tackle this we'd also have to consider how cert generation and importing is done differently across different operating systems. Finally, we'd probably only want to do this once per local dev environment - and not once per arc app. My naive off-the-top-of-my-head idea for this is to do this song and dance as part of an arc post-install script (something like generate the certs and store in a `~/.arc` folder and, if possible, do the root cert importing for the developer).

2.  Or we say 'fuck it' to trying to handle this for the developer and accept the manual overhead of having devs accept cert warnings in their browser when using sandbox. Maybe `npx sandbox` generates a key pair at runtime if it can't find any? And do we need to write these to disk? Maybe we can simply store them in memory? This would decrease littering app projects with extra files.
